ADR/0032-decoupling-deployment.md

```diff
@@ -0,0 +1,211 @@
+  that accept a kubeconfig and snapshot as they do now, but those pipelines should now take initial
+  steps to *deploy the instance of their application to be tested* to the deployment target provided
+  by [integration-service] using the provided kubeconfig.
+- We should promote [release-service] as the primary means to advertise to [renovatebot] that one or
+  more images have passed testing and are ready to be promoted to a particular environment (with
+  a lowercase "e") by way of image tags in a registry.
+
+### Out of scope
+
+Some other interesting ideas that are floating around these days, but which should be taken up in
+other ADRs, if we take them up at all:
+
+- Decoupling [integration-service] and [release-service] from a shared [Snapshot] API. They will
+  continue to share a [Snapshot] API as of this ADR.
+- Decoupling [integration-service] from the [application-service] APIs, like [Application] and
+  [Component]. It will still promote images to the "global candidate list" (on the [Component]
+  resources) and continue to use the list of [Components] to guide its construction of [Snapshots].
+
+### Use Case Descriptions
+
+**During onboarding**: whereas today when a user requests a new appstudio tier namespace, the tier
+template includes an [Environment] that the integration-service will promote to. Tomorrow, the
+appstudio tier template should no longer include an [Environment] which on its own will cause
+integration-service to _not_ trigger a deployment when testing completes. Instead, the appstudio
+tier template should include a [ReleasePlan] with a reference to the [push-to-registry]
+release pipeline. This new default [ReleasePlan] should carry parameters such that whenever
+a [Snapshot] is successfully tested, a [Release] is created that re-tags the images in build-time
+quay repositories with a tag like `:released` or `:validated` (name tbd). The
+[push-to-registry] pipeline can use the `appstudio-pipeline` service account in the user's
+dev workspace, which already has push access to the repository in question.
+
+```mermaid
+flowchart TD
+    NSTierTemplate --> |provides| ReleasePlan
+    A[fa:fa-code-commit Git Commit] --> |webhook| PipelinesAsCode
+    PipelinesAsCode --> |creates| build[Build PipelineRun]
+    build --> |triggers| integration-service
+    integration-service --> |asks| decision{Is there a ReleasePlan?}
+    ReleasePlan --> |influences| decision
+    decision --> |yes| Release
+    build --> |pushes the image| quay.io/redhat-user-workloads/tenant/component
+    Release --> |tags the image for release| quay.io/redhat-user-workloads/tenant/component
+    quay.io/redhat-user-workloads/tenant/component --> renovatebot[User provided RenovateBot]
+    renovatebot --> |updates| gitops-repo[User provided GitOps repo]
+    gitops-repo --> |causes deployment via| argocd[User provided ArgoCD]
+    argocd --> |deploys to| cluster[User provided cluster]
```

Comment:
Maybe emphasis that ArgoCD is just an exmaple and other gitops tools can be used. It can even update an Helm repo with the new images, the options are limitless.

Reply:
Updated in 382bcb9.